scipy/spatial/setup.py

```diff
@@ -105,7 +106,15 @@ def configuration(parent_package='', top_path=None):
                                **numpy_nodepr_api)
     ext._pre_build_hook = pre_build_hook
 
-    config.add_extension('_voronoi',
+    if int(os.environ.get('SCIPY_USE_PYTHRAN', 1)):
+        import pythran
+        ext = pythran.dist.PythranExtension(
+            'scipy.spatial._voronoi_pythran',
```

Comment:
I think you may want `scipy.spatial._voronoi` here to shadow the Cython extension like Ralf was saying, assuming there's a way to drop-in replace without modifying the Cython code return value (`None`).